assets/js/components/ModalDialog.js

```diff
@@ -0,0 +1,145 @@
+/**
+ * ModalDialog component.
+ *
+ * Site Kit by Google, Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { useInstanceId } from '@wordpress/compose';
+import { createInterpolateElement } from '@wordpress/element';
+import { sprintf, __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import {
+	SpinnerButton,
+	Dialog,
+	DialogTitle,
+	DialogContent,
+	DialogFooter,
+} from 'googlesitekit-components';
+import Link from './Link';
+
+const ModalDialog = ( {
+	dialogActive,
+	handleDialog,
+	title,
+	provides,
+	handleConfirm,
+	subtitle,
+	confirmButton,
+	dependentModules,
+	danger,
+	inProgress = false,
+} ) => {
+	const instanceID = useInstanceId( ModalDialog );
+	const describedByID = `googlesitekit-dialog-description-${ instanceID }`;
+	const hasProvides = !! ( provides && provides.length );
+
+	return (
+		<Dialog
+			open={ dialogActive }
+			aria-describedby={ hasProvides ? describedByID : undefined }
+		>
+			<DialogTitle>{ title }</DialogTitle>
+			{ subtitle && <p className="mdc-dialog__lead">{ subtitle }</p> }
```

Comment:
Although outside of `DialogContent` this retains the existing layout, and should be compatible with the `@material/web` dialog component, we'll probably have to add a new `DialogSubtitle` component for it, but it will be able to go in the [`headline-suffix` slot](https://github.com/material-components/material-web/blob/2ef3606bfe66fd700d803812b524b95c22ea4028/dialog/lib/dialog.ts#L285), again outside of the content area.

Reply:
Looks like this subtitle causes storybook issues here: https://google.github.io/site-kit-wp/storybook/develop/?path=/story/components-confirmdisconnect--confirm-disconnect-without-features